### ISSUE TITLE
your GAN network is not working!!!

### ISSUE THREAD

**Issue by Johnson-yue** [hologerry/AGIS-Net#10]
Hi, I train your network with your default configuration and skeleton_gray_color for pretrain, skeleton_gray_texture is for finetune. and I got some result as follow:

1. In pretrain and finetune process, the **G_GAN** and **G_GAN_B** never goes down , it always around 100 . 
2. the result of generate  both __pretrain__ and __finetune__ is always blur. 

So, I have reason believe that you GAN is not working and  the result of your paper is mostly depend on pretrain performance!

here is my train process log, if your log is  different from me, I will check it again.

and

**Comment by hologerry** [comment]
The original GAN loss in our model is indeed not such useful comparing to L1 and Lcx, but we can not deny its role.
I never said the pre-train procedure is not important.
If you doubt any part of the model, you can perform the ablation studies.
The absolute value of losses may not vary much during the late training processes, especially for the GAN loss which indicating the quality at a high level, in glyph related tasks as we discussed in the paper.
Where is the training process log?
OK, your understanding is right.

**Comment by Johnson-yue** [comment]
![train_log_gan](https://user-images.githubusercontent.com/10268274/68574120-2b8c8b00-04a4-11ea-9c14-4d42d399e3be.png)
1 ) This is finetune log， and train log is here , the G_GAN and G_GAN_B is very stable.

2) I agree with you about pretrain is important , and I never said it is problem, I just say it. 

3)  Even In the early stages of training， the GAN losses is never goes down a lot。

4） As your said， and I check your ablation stduies， you only compare with and without L_local ，not compare with and without L_adv， Maybe you can remove **Shape Discriminator** and **Texture   Discriminator**  for your ablation studies。

**Comment by hologerry** [comment]
I mean you can perform the ablation studies, if you doubt these two discriminators.
You can even write something comparing the GAN with "No Discriminator GAN". haha
We did not present this part in paper because they are not novel, they are just PathGAN like discriminators.
